src/tests/functional/inference_engine/transformations/common_optimizations/gru_cell_fusion.cpp

```diff
@@ -0,0 +1,160 @@
+        params.push_back(std::dynamic_pointer_cast<Parameter>(Bh));
+    }
+
+    auto axis_0 = make_shared<Constant>(i64, ov::Shape{}, 0);
+    auto axis_1 = make_shared<Constant>(i64, ov::Shape{}, 1);
+    auto split_lenghts = make_shared<Constant>(i64, ov::Shape{2}, vector<size_t>{input_size, hidden_size});
+    auto split_WRzr = make_shared<VariadicSplit>(WRzr, axis_1, split_lenghts);
+    auto split_WRh = make_shared<VariadicSplit>(WRh, axis_1, split_lenghts);
+    auto Wzrh = make_shared<Concat>(ov::OutputVector{split_WRzr->output(0), split_WRh->output(0)}, 0);
+    auto Rzrh = make_shared<Concat>(ov::OutputVector{split_WRzr->output(1), split_WRh->output(1)}, 0);
+
+    auto B = make_shared<Concat>(ov::OutputVector{Bzr, Bh}, 1);
+
+    auto squeeze_B = make_shared<Squeeze>(B, axis_0);
+    auto cell = make_shared<GRUCell>(X, H, Wzrh, Rzrh, squeeze_B, hidden_size, vector<string>{activation_1, activation_2});
+    return std::make_shared<ngraph::Function>(ov::OutputVector {cell}, params);
+}
+} // namespace
+
+struct GRUFusionParams {
+    string activation_1;
+    string activation_2;
+    size_t batch;
+    size_t hidden_size;
+    size_t input_size;
+    bool use_bias_add_1;
+    bool use_bias_add_2;
+};
+
+class GRUFusionTest
+        : public testing::WithParamInterface<GRUFusionParams>,
+          public TransformationTestsF {
+};
+
+TEST_P(GRUFusionTest, GRUCellPattern) {
+    auto p = GetParam();
+    {
+        function = gen_model(p.activation_1, p.activation_2, p.batch,
+                             p.hidden_size, p.input_size, p.use_bias_add_1, p.use_bias_add_2, false);
+        manager.register_pass<ov::pass::GRUCellFusion>();
+    }
+
+    {
+        function_ref = gen_reference(p.activation_1, p.activation_2, p.batch,
+                                     p.hidden_size, p.input_size, p.use_bias_add_1, p.use_bias_add_2);
+    }
+    comparator.enable(FunctionsComparator::CmpValues::ACCURACY);
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+    comparator.enable(FunctionsComparator::CmpValues::ATTRIBUTES);
+}
+
+class GRUFusionTestDyn
+        : public testing::WithParamInterface<GRUFusionParams>, public TransformationTestsF {
+};
+
+TEST_P(GRUFusionTestDyn, GRUCellPatternDynamicShapes) {
+    auto p = GetParam();
+    {
+        function = gen_model(p.activation_1, p.activation_2, p.batch, p.hidden_size, p.input_size, false, false, true);
+        manager.register_pass<ov::pass::GRUCellFusion>(); // the transformation won't be applied
+    }
+}
+
+static std::vector<GRUFusionParams> params = {
+        GRUFusionParams{"sigmoid", "tanh", 1, 1, 1, true, true},
+        GRUFusionParams{"tanh", "sigmoid", 2, 128, 32, true, true},
+        GRUFusionParams{"tanh", "tanh", 2, 128, 32, true, false},
+        GRUFusionParams{"sigmoid", "relu", 2, 128, 32, false, false},
+        // accuracy issue, it looks like the test infrastructure issue, the graphs are the same.
```

Comment:
@itikhono Do you plan to track it in a ticket later or are you already on it?

Reply:
I will create a new ticket

Reply:
I think will be better to add ticket number in the comment

Reply:
issue 92424